akka-cluster/src/test/scala/akka/cluster/ClusterLogSpec.scala

```diff
@@ -25,33 +25,82 @@ object ClusterLogSpec {
 
 }
 
-class ClusterLogSpec extends AkkaSpec(ClusterLogSpec.config) with ImplicitSender {
+abstract class ClusterLogSpec(config: Config) extends AkkaSpec(config) with ImplicitSender {
 
-  val selfAddress: Address = system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+  def this(s: String) = this(ConfigFactory.parseString(s))
 
-  val cluster = Cluster(system)
-  def clusterView: ClusterReadView = cluster.readView
+  protected val selfAddress: Address = system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+
+  protected val cluster = Cluster(system)
+
+  protected def clusterView: ClusterReadView = cluster.readView
+
+  protected def awaitUp(): Unit = {
+    awaitCond(clusterView.isSingletonCluster)
+    clusterView.self.address should ===(selfAddress)
+    clusterView.members.map(_.address) should ===(Set(selfAddress))
+    awaitAssert(clusterView.status should ===(MemberStatus.Up))
+  }
+
+  /** The expected log info pattern to intercept after a `cluster.join`. */
+  protected def join(expected: String): Unit =
+    EventFilter.
+      info(occurrences = 1, pattern = expected).
+      intercept(cluster.join(selfAddress))
+
+  /** The expected log info pattern to intercept after a `cluster.down`. */
+  protected def down(expected: String): Unit =
+    EventFilter.
+      info(occurrences = 1, pattern = expected).
+      intercept(cluster.down(selfAddress))
+}
+
+class ClusterLogDefaultSpec extends ClusterLogSpec(ClusterLogSpec.config) {
 
   "A Cluster" must {
 
     "Log a message when becoming and stopping being a leader" in {
-      EventFilter
-        .info(occurrences = 1, pattern = "is the new leader")
-        .intercept {
-          cluster.join(selfAddress)
-        }
-
-      awaitCond(clusterView.isSingletonCluster)
-      clusterView.self.address should ===(selfAddress)
-      clusterView.members.map(_.address) should ===(Set(selfAddress))
-      awaitAssert(clusterView.status should ===(MemberStatus.Up))
-
-      EventFilter
-        .info(occurrences = 1, pattern = "is no longer the leader")
-        .intercept {
-          cluster.down(selfAddress)
-        }
+      join("is the new leader")
+      awaitUp()
+      down("is no longer the leader")
     }
+  }
+}
+
+abstract class ClusterLogVerboseSpec(config: Config) extends ClusterLogSpec(config) {
 
+  def this(s: String) = this(ConfigFactory.parseString(s))
```

Comment:
This helper constructor exists in `AkkaSpec` so a string can be passed all the way down.

Reply:
Or all the way "up"

Reply:
Thanks @chbatey I was wanting to put the milestone, just didn't know what we're working on yet 👍

Reply:
@chbatey true but this is for a new abstract spec that needs it in order to allow one of the two new specs, the default behavior (off) to be the same as the LogInfo spec, which I think is cleaner, and only the verbose (on) spec to pass in Config to the new abstract class vs string.